New product documentation

ggbecker opened the pull request:
Introduce documentation on how to create a new product in the project.

Also contains simple bash script to help creating the directory structure and some of the needed files.

Relates to #4272

jan-cerny commented on example/README.md:
```diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
+
+Follow these steps to create a new product in the project:
+
+1. Create a new folder in the root directory which will holds the files related to your new product. To illustrate the process we will use the name `custom6` which basically means that the product is called `custom` and the major version is `6`.
```
holds -> hold

ggbecker replied:
fixed

jan-cerny commented on example/README.md:
```diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
+if (SSG_PRODUCT_DEBIAN8)
+    add_subdirectory("debian8")
+endif()
+if (SSG_PRODUCT_CUSTOM6
```
missing )

ggbecker replied:
fixed

jan-cerny commented on example/README.md:
````diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
+    add_subdirectory("eap6")
+endif()
+...
+```
````
they also should add it in `./build_product`

ggbecker replied:
I'll add this instruction as well. Thanks.

ggbecker replied:
Added.

jan-cerny commented on example/README.md:
````diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
+```
+...
+option(SSG_PRODUCT_DEBIAN8 "If enabled, the Debian 8 SCAP content will be built" TRUE)
+option(SSG_PRODUCT_CUSTOM6 "If enabled, the Custom 6 SCAP content will be built" TRUE)
````
I suggest doing this in a "diff"-like format so that it will obvious they will add only the line with custom

ggbecker replied:
I have changed to make the addition bold instead. What do you think about?

I didn't find an easy way to make it "diff"-like, do have any suggestions? I would find painful to put minus (-) and plus (+) manually.

ggbecker replied:
Assuming that setting text to `bold` is enough. Resolving conversation.

ggbecker commented:
I have addressed review requests. But the documentation is still not complete. Further work needs to be done.

ggbecker commented:
Added four more items to the documentation. Still in progress.

ggbecker commented:
More sections added. I looks that only few items are missing now, mainly in the `ssgconstants.py` file.

ggbecker commented:
- Added remaining sections for xslt transformation files
- Added script which create every file needed for a new product but files which need manual editing.

It is ready to review.

Note: the script is there in a best effort case, it lacks documentation on how to use it but it might help. Maybe it can evolve to something more useful.

jan-cerny commented:
@ggbecker This is a great contribution!
